server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -835,432 +835,465 @@ public void deleteSnapshots(
         long repositoryDataGeneration,
         IndexVersion repositoryFormatIndexVersion,
         SnapshotDeleteListener listener
+    ) {
+        createSnapshotsDeletion(snapshotIds, repositoryDataGeneration, repositoryFormatIndexVersion, new ActionListener<>() {
+            @Override
+            public void onResponse(SnapshotsDeletion snapshotsDeletion) {
+                snapshotsDeletion.runDelete(listener);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                listener.onFailure(e);
+            }
+        });
+    }
+
+    private void createSnapshotsDeletion(
+        Collection<SnapshotId> snapshotIds,
+        long repositoryStateId,
+        IndexVersion repositoryMetaVersion,
+        ActionListener<SnapshotsDeletion> listener
```

Comment:
This might look a bit weird, you're probably wondering why we do this in a separate method from the one which calls `runDelete()`. It will become clearer in a future PR which will re-use this for the `cleanup()` process too, since today `cleanup()` is effectively a delete of an empty set of snapshots, it does the same cleanup process and needs basically the same context.